src/parse.js

```diff
@@ -391,11 +423,16 @@ Parser.prototype.parseValueRecordList = function() {
     return values;
 };
 
-Parser.prototype.parsePointer = function(description) {
+Parser.prototype.parsePointer = function(description, storeOffset = false) {
     const structOffset = this.parseOffset16();
     if (structOffset > 0) {
         // NULL offset => return undefined
-        return new Parser(this.data, this.offset + structOffset).parseStruct(description);
+        const offset = this.offset + structOffset;
+        const struct = new Parser(this.data, offset).parseStruct(description);
```

Comment:
Why do we need to create a new ``Parser`` object here when we're already in one? A better solution might be to add a stack to the parser that we can push the current cursor location onto and jump to a new cursor location and then when we're done there we pop it off to jump back to the original cursor location. Would also reduce the amount of memory allocations used when parsing fonts, speeding up the font parsing.